test/fuzz/lua/test_engine.lua

```diff
@@ -0,0 +1,1411 @@
+-- becomes "%", "%[" becomes "[", etc.
+local err_pat_whitelist = {
+    -- Multi-engine transactions aren't supported, see
+    -- https://github.com/tarantool/tarantool/issues/1958 and
+    -- https://github.com/tarantool/tarantool/issues/1803.
+    "Can not perform index build in a multi-statement transaction",
+    -- DDL on a space is locked until the end of the current DDL
+    -- operation.
+    "the space is already being modified",
+    -- The test actively uses transactions that concurrently
+    -- changes a data in a space, this can lead to errors below.
+    "Transaction has been aborted by conflict",
+    "Vinyl does not support rebuilding the primary index of a non%-empty space",
+    "fiber is cancelled",
+    "fiber slice is exceeded",
+    "A multi%-statement transaction can not use multiple storage engines",
+    "Can not perform index build in a multi%-statement transaction",
+    "Index '[%w_]+' %(HASH%) of space '[%w_]+' %(memtx%) does not support pagination",
+    "Can't create or modify index '[%w_]+' in space '[%w_]+': primary key must be unique",
+    "Can't create or modify index '[%w_]+' in space '[%w_]+': hint is only reasonable with memtx tree index",
+    "Get%(%) doesn't support partial keys and non%-unique indexes",
+    "Index '[%w_]+' %(RTREE%) of space '[%w_]+' %(memtx%) does not support max%(%)",
+    "Index '[%w_]+' %(RTREE%) of space '[%w_]+' %(memtx%) does not support min%(%)",
+    -- Blocked by tarantool#10262.
+    "attempt to index local 'tuple' %(a nil value%)",
+}
+
+local function keys(t)
+    assert(next(t) ~= nil)
+    local table_keys = {}
+    for k, _ in pairs(t) do
+        table.insert(table_keys, k)
+    end
+    return table_keys
+end
+
+local function rmtree(s)
+    log.info(('CLEANUP %s'):format(s))
+    if (fio.path.is_file(s) or fio.path.is_link(s)) then
+        fio.unlink(s)
+        return
+    end
+    if fio.path.is_dir(s) then
+        for _, i in pairs(fio.listdir(s)) do
+            rmtree(s..'/'..i)
+        end
+    end
+end
+
+local function rand_char()
+    return string.char(math.random(97, 97 + 25))
+end
+
+local function rand_string(length)
+    length = length or 10
+    local res = ''
+    for _ = 1, length do
+        res = res .. rand_char()
+    end
+    return res
+end
+
+local function oneof(tbl)
+    assert(type(tbl) == 'table')
+    assert(next(tbl) ~= nil)
+
+    local n = table.getn(tbl)
+    local idx = math.random(1, n)
+    return tbl[idx]
+end
+
+local function unique_ids(max_num_ids)
+    local ids = {}
+    for i = 1, max_num_ids do
+        table.insert(ids, i)
+    end
+    return function()
+        local id = math.random(#ids)
+        local v = ids[id]
+        assert(v)
+        table.remove(ids, id)
+        return v
+    end
+end
+
+-- Forward declaration.
+local index_create_op
+
+local function random_int()
+    return math.floor(math.random() * 10^12)
+end
+
+local RTREE_DIMENSION = oneof({2, 4, 8})
+
+local function random_array()
```

Comment:
Why is this called `random_array()`? There are no `math.random()` like in `random_map()` bellow.

Reply:
Actually `random_array` uses random dimension `RTREE_DIMENSION` that *generated* only once at start, see above.

I've updated the function a bit:

```diff
--- a/test/fuzz/lua/test_engine.lua
+++ b/test/fuzz/lua/test_engine.lua
@@ -181,10 +181,16 @@ local function random_int()
     return math.floor(math.random() * 10^12)
 end
 
-local RTREE_DIMENSION = oneof({2, 4, 8})
+-- Maximal possible R-tree dimension,
+-- see <src/lib/salad/rtree.h>.
+local RTREE_MAX_DIMENSION = 20
 
+local RTREE_DIMENSION = math.random(RTREE_MAX_DIMENSION)
+
+-- RTREE is a single index that support arrays, length of arrays
+-- depends on a RTREE's dimension.
 local function random_array()
-    local n = RTREE_DIMENSION
+    local n = RTREE_DIMENSION * 2
     local arr = {}
     for i = 1, n do
         table.insert(arr, i)
```